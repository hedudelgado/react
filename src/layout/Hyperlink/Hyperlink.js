import React from 'react';
import {
  node,
  oneOf,
  string,
} from 'prop-types';
import classNames from 'classnames';

const Hyperlink = ({
  variant,
  href,
  children,
  className,
  ...otherProps
}) => {
  const classes = classNames(
    {
      'wb-btn wb-btn--full-width wb-btn--small max-fit': (variant === 'primary' || variant === 'secondary'),
      'wb-btn--primary': variant === 'primary',
      'wb-btn--secondary': variant === 'secondary',
    },
    className,
  );

  return (
    <a href={`${href}`} className={classes} {...otherProps}>
      {children}
    </a>
  );
};

Hyperlink.propTypes = {
  /**
   * Variant defines the Hyperlink's visual represenation.
   */
  variant: oneOf(['regular', 'primary', 'secondary']),
  /**
   * The anchor tag's target URL.
   */
  href: string,
  /**
   * The Hyperlink contents. It is usually a text, but it can also be further DOM.
   */
  children: node,
  /**
   * Extra CSS classes the component might need.
   */
  className: string,
};

Hyperlink.defaultProps = {
  variant: 'regular',
  href: '#',
  children: null,
  className: '',
};

export default Hyperlink;
