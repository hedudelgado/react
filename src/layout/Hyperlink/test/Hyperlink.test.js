import React from 'react';
import { shallow } from 'enzyme';
import Hyperlink from '../Hyperlink';

describe('Hyperlink component', () => {
  const variants = ['primary', 'secondary'];

  it('should render the component', () => {
    const props = {
      variant: 'regular',
      href: 'https://www.premierinn.com',
    };
    const wrapper = shallow(<Hyperlink {...props}>This is a hyperlink</Hyperlink>);

    expect(wrapper).toMatchSnapshot();
  });

  it.each(variants)('should render the %s variation', (variant) => {
    const wrapper = shallow(<Hyperlink variant={variant}>This is another link</Hyperlink>);
    expect(wrapper).toHaveClassName(`wb-btn--${variant}`);
  });
});
