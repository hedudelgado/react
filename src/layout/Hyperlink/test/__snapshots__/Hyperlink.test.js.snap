// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hyperlink component should render the component 1`] = `
<a
  className=""
  href="https://www.premierinn.com"
>
  This is a hyperlink
</a>
`;
